crypto/BRCryptoWalletManager.c

```diff
@@ -934,38 +998,68 @@ cryptoWalletManagerHandleTransferGEN (BRCryptoWalletManager cwm,
     // TODO: Determine the currency from `transferGeneric`
     BRCryptoCurrency currency = cryptoNetworkGetCurrency (cwm->network);
 
-    BRCryptoUnit unit = cryptoNetworkGetUnitAsBase (cwm->network, currency);
-    BRCryptoUnit unitForFee = cryptoNetworkGetUnitAsBase (cwm->network, currency);
-
-    // Create the generic transfers...
-    BRCryptoTransfer transfer = cryptoTransferCreateAsGEN (unit, unitForFee, transferGeneric);
-
-    // TODO: Determine the wallet from transfer/transferGeneric
-    BRCryptoWallet   wallet   = cwm->wallet;
-
-    // .. and announce the newly created transfer.
-    cwm->listener.transferEventCallback (cwm->listener.context,
-                                         cryptoWalletManagerTake (cwm),
-                                         cryptoWalletTake (wallet),
-                                         cryptoTransferTake(transfer),
-                                         (BRCryptoTransferEvent) {
-                                             CRYPTO_TRANSFER_EVENT_CREATED
-                                         });
-
-    // Add the restored transfer to its wallet...
-    cryptoWalletAddTransfer (cwm->wallet, transfer);
+    // TODO: Determine the wallet from transferGeneric (not always the primaryWallet)
+    BRCryptoWallet wallet = cwm->wallet;
+
+    // Look for a known transfer
+    BRCryptoTransfer transfer = cryptoWalletFindTransferAsGEN (wallet, transferGeneric);
+
+    // If we don't know about `transferGeneric`, create a crypto transfer
+    if (NULL == transfer) {
+        BRCryptoUnit unit = cryptoNetworkGetUnitAsBase (cwm->network, currency);
+        BRCryptoUnit unitForFee = cryptoNetworkGetUnitAsBase (cwm->network, currency);
+
+        // Create the generic transfer...
+        transfer = cryptoTransferCreateAsGEN (unit, unitForFee, transferGeneric);
```

Comment:
Leaked?

Reply:
I don't think so.  `transfer` has a non-NULL  value; the end of the function as `cryptoTransferGive(transfer)`